examples/plot_metric_examples.py

```diff
@@ -0,0 +1,410 @@
+# You'll notice that the points from the same labels are closer together,
+# but they are not necessary in a same cluster. This is particular to LMNN
+# and we'll see that some other algorithms implicitly enforce points from
+# the same class to cluster together.
+# 
+# You can find the paper
+# `here <http://jmlr.csail.mit.edu/papers/volume10/weinberger09a/weinberger09a.pdf>`__.
+# 
+
+
+######################################################################
+# Fit and then transform!
+# -----------------------
+# 
+
+# setting up LMNN
+lmnn = metric_learn.LMNN(k=5, learn_rate=1e-6)
+
+# fit the data!
+lmnn.fit(X, Y)
+
+# transform our input space
+X_lmnn = lmnn.transform(X)
+
+
+######################################################################
+# So what have we learned? The matrix :math:`M` we talked about before.
+# Let's see what it looks like.
+# 
+
+lmnn.metric()
+
+
+######################################################################
+# Now let us plot the transformed space - this tells us what the original
+# space looks like after being transformed with the new learned metric.
+# 
+
+plot(X_lmnn, Y)
+
+
+######################################################################
+# Pretty neat, huh?
+# 
+# The rest of this notebook will briefly explain the other Metric Learning
+# algorithms before plotting them. Also, while we have first run ``fit``
+# and then ``transform`` to see our data transformed, we can also use
+# ``fit_transform``. The rest of the examples and illustrations will use
+# ``fit_transform``.
+# 
+
+
+######################################################################
+# Information Theoretic Metric Learning
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# 
+# ITML uses a regularizer that automatically enforces a Semi-Definite
+# Positive Matrix condition - the LogDet divergence. It uses soft
+# must-link or cannot like constraints, and a simple algorithm based on
+# Bregman projections. Unlike LMNN, ITML will implicitly enforce points from
+# the same class to belong to the same cluster, as you can see below.
+# 
+# Link to paper:
+# `ITML <http://www.cs.utexas.edu/users/pjain/pubs/metriclearning_icml.pdf>`__.
+# 
+
+itml = metric_learn.ITML_Supervised(num_constraints=200)
+X_itml = itml.fit_transform(X, Y)
+
+plot(X_itml, Y)
+
+
+######################################################################
+# Sparse Determinant Metric Learning
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# 
+# Implements an efficient sparse metric learning algorithm in high
+# dimensional space via an :math:`l_1`-penalised log-determinant
+# regularization. Compare to the most existing distance metric learning
+# algorithms, the algorithm exploits the sparsity nature underlying the
+# intrinsic high dimensional feature space.
+# 
+# Link to paper here:
+# `SDML <http://lms.comp.nus.edu.sg/sites/default/files/publication-attachments/icml09-guojun.pdf>`__.
+# 
+# One feature which we'd like to show off here is the use of random seeds.
```

Comment:
not sure this makes much sense here? perhaps remove this paragraph? or keep it but without mentioning constraints specifically (just randomness in the algorithm)

Reply:
I agree, the simpler is to remove it I guess, since we'll also change this and put the random seed in init (see #171)